### ISSUE TITLE
PER-CS v2 support

### ISSUE THREAD

**Issue by Wirone** [PHP-CS-Fixer/PHP-CS-Fixer#7247]
After #6707 was merged we can start working on `@PER-CS2.0`. Initial work has been done in #6547 but it's massively outdated now, that's why I am going to close that PR and move checklist here as a base for further works (should be verified).

- [ ] create `@PER-CS2.0` (and `@PER-CS2.0:risky` if needed) and mark them as experimental/internal (whatever, just let people know it's not production-ready)
- [ ] `protected_to_private`, but for enums only
- [ ] `no_singleline_whitespace_before_semicolons`, but for arrow functions only
- [ ] #6518
- [x] #6529
- [ ] `fn` and `=>` must be surrounded by a space
- [ ] In multiline arrow functions, `=>` must be placed after a new line
- [ ] `=>` must be indented once when placed after a new line
- [ ] Method chaining must be fully multiline, with the first item on a new line
- [ ] when everything is done, mark rule set(s) as generally available

Each checklist item should be provided as separate PR, so review process is easier and things may go forward

**Comment by Crell** [comment]
We'll also need an enum casing rule to enforce PascalCase.

**Comment by Crell** [comment]
First step here: #7249

**Comment by SamMousa** [comment]
I think short array syntax is missing from the PER-CS 2.0 set.

https://www.php-fig.org/per/coding-style/#11-arrays

> Arrays MUST be declared using the short array syntax.

**Comment by SpacePossum** [comment]
Thanks @SamMousa , added to this PR https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8171 :+1:  :crossed_fingers:

**Comment by Kenneth-Sills** [comment]
> Why is [@PER-CS2.0:risky](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/doc/ruleSets/PER-CS2.0Risky.rst) based on [@PER-CS1.0:risky](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/doc/ruleSets/PER-CS1.0Risky.rst), instead of on [@PER-CS2.0](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/doc/ruleSets/PER-CS2.0.rst) + risky rules?

Oh, I think there's a miscommunication here. `risky` rule sets *only* include risky rules. You need to enable both the base and the risky rule sets at the same time.

Take a look at every `Risky` set [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/tree/bb0a630461dcdf33302422303e6bb61db5cd50c5/src/RuleSet/Sets), you'll seee the same thing. 😄

**Comment by kubawerlos** [comment]
Exactly, all the sets without `:risky` in the name are considered non-risky and have only non-risky rules.

**Comment by oliverklee** [comment]
FYI, version 3 of the PHP-FIG PER-Coding Style guidelines have been released: https://www.php-fig.org/per/coding-style/

There's also a document on migrating from PER-CS v2.0 to PER-CS v3.0: https://github.com/php-fig/per-coding-style/blob/master/migration-3.0.md

**Comment by keradus** [comment]
FYI https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8841

also, consider to contribute to cover missing style rules for 3.0 , if possible 🙇🏻
